_commands/tools/grep.md

```diff
@@ -3,22 +3,100 @@
 
 grep
 -------
-TODO: Add documentation for this command by submitting a pull request.
-<!-- one line explanation would go here -->
 
-<!-- minimal example -->
+`grep` is a program for searching text files for lines that match regular exprssions. It can be used for all sorts of pattern-matching and text-based query analysis.
+
+<!--more-->
+
 ~~~ bash
-*short* example of command usage and output
+$ grep "wolf" animals.txt
```

Comment:
Where did this `animals.txt` file come from? Providing a link to that file or a way to get it would be helpful for people to follow along with.

Reply:
Well originally it was just hypothetical/conceptual, but I agree that it would be easier to follow along with an actual file and actual output. I've included an animals.txt file, linked to it from the page, and redesigned the examples to be meaningful and real-world.